Improve kernel smoothing (#2701)
openturns/openturns: python/doc/theory/data_analysis/kernel_smoothing.rst

- this last kernel smoothed PDF is truncated within the initial range :math:`[min, max]` (conditional PDF).

Log-transform treatment
~~~~~~~~~~~~~~~~~~~~~~~

In this section, we consider a random variable i.e. :math:`d = 1`. This treatment is highly suited to skewed distributions,
which are all challenging for kernel smoothing. See [charpentier2015]_ to get more details.

> We denote by :math:`(X_i)_{1 \leq i  \leq n}` some independent random variates, identically distributed according to :math:`X`.

Reviewer: ```suggestion
Let :math:`(X_i)_{1 \leq i  \leq \sampleSize}` be some independent random variates, identically distributed according to :math:`X`.
```

Author: fixed

The log-transform treatment maps each :math:`X_j` into :math:`Y_j` as follows:

.. math::

    Y_j  = T(X_j) = \left |

>     \begin{array}{ll}
>        \log (X_j - \min_{i} X_i + \delta) & \mbox{if } \gamma_1(X) >0\\
>        \log (\max_{i} X_i - X_j + \delta) & \mbox{if } \gamma_1(X) >0
>        \end{array}
>        \right.

Reviewer: Please use `cases`, which is designed for this purpose (see [cases](https://www.overleaf.com/latex/examples/cases/nndqpbymnchn)):

```suggestion
    \begin{cases}
       \log (X_j - \min_{i} X_i + \delta) & \mbox{if } \gamma_1(X) >0\\
       \log (\max_{i} X_i - X_j + \delta) & \mbox{if } \gamma_1(X) >0
    \end{cases}
    \right.
```

Author: fixed
